### PR TITLE
Add XML declaration custom single/double quote style support in glif and *.plist files

### LINKED ISSUE #156: Allow customizing quote char in xml header
- [x] add a `quote_char` method to `WriteOptions`
- [x] When serializing the `.glif`s, (which we serialize manually) write the header to the stream ourselves, using the specified character
- [x] When serializing plists, we don't control writing the header, so we'll have to rewrite. There are options here. All `plist` writing is now handled through a method in `write.rs`. 
  - One option is to write to disk, and then use the unix/windows specific methods to go and overwrite the header in place, before closing the file handle: see [`std::os::unix::fs::FileExt`](http://doc.rust-lang.org/1.53.0/std/os/unix/fs/trait.FileExt.html) and [`std::os::windows::fs::FileExt`](http://doc.rust-lang.org/1.53.0/std/os/windows/fs/trait.FileExt.html); you'll need slightly different impls on either platform.
  - the other option would be to have a custom `Write` type, that would wrap or replace the `BufWriter`; this would intercept the header and replace it, if needed, otherwise forwarding the bytes to the inner writer.

Thinking about this, I think the first option is totally reasonable. The second option is probably marginally more 'efficient', but I don't think it matters; the main thing we want to avoid is closing and reopening all the file handles.

_Originally posted by @cmyr in https://github.com/linebender/norad/issues/148#issuecomment-893514934_

### PR DESCRIPTION
**Pull request by chrissimpkins**
Closes #156

### CODE REVIEW

File: src/layer.rs [hunk 1]
```diff
@@ -349,19 +349,14 @@ impl Layer {
 
         self.layerinfo_to_file_if_needed(path, opts)?;
 
-        for (name, glyph_path) in self.contents.iter() {
```
Comment (chrissimpkins) [inline src/layer.rs:352]: It looks like this is duplicate code for the glif file writes and should be removed elsewhere if this PR is not merged

File: src/write.rs [hunk 2]
```diff
@@ -36,14 +41,15 @@ impl Default for WriteOptions {
             xml_opts: Default::default(),
             whitespace_char: b'\t',
             whitespace_count: 1,
+            quote_style: QuoteStyle::Double,
```
Comment (chrissimpkins) [inline src/write.rs:44]: Keep double quotes as default approach

**Comment by chrissimpkins** [comment]
Ready for review.  I'll begin working on tests.

File: src/write.rs [hunk 3]
```diff
@@ -69,21 +81,39 @@ impl WriteOptions {
...
     pub fn xml_options(&self) -> &XmlWriteOptions {
         &self.xml_opts
     }
 }
 
+#[derive(Debug, Clone)]
+pub enum QuoteStyle {
+    Single,
+    Double,
+}
```
[the remark this answers is not in the document]
Comment (chrissimpkins) [reply src/write.rs]: Thanks Colin!

I accepted this commit and then refactored all uses in https://github.com/linebender/norad/pull/157/commits/8b59d3fde05d68adf7d8c8754cdc24f1656c4128

**Comment by chrissimpkins** [comment]
Do we want to add Win CI testing since there are platform-specific approaches added here?

File: src/layer.rs (hunk 1 above)
Comment (chrissimpkins) [reply src/layer.rs:352]: > I thiiiink what I'd like to do is revert that previous commit, and then I'll redo it and redo my measurements. I'll do that (the reverting) and then you'll need to patch up this PR.

Np, will do

**Comment by madig** [comment]
I think so!

**Comment by chrissimpkins** [comment]
> I think so!

https://github.com/linebender/norad/pull/161

**Comment by chrissimpkins** [comment]
Tests added

File: src/write.rs [hunk 4]
```diff
@@ -69,21 +81,44 @@ impl WriteOptions {
         self
     }
 
+    /// Builder-style method to customize the XML declaration attribute definition quote
+    /// char.
+    ///
+    /// By default, we indent with double quotes.
+    ///
+    /// The quote style is defined with a [QuoteChar] enum argument.
```
Comment (cmyr) [inline src/write.rs:89]:
```suggestion
    /// The quote style is defined with a [`QuoteChar`] enum argument.
```

**Review by cmyr — APPROVED** [review]
Looks good, one tiny fixup, I'll apply that and merge. :)

**Pull request merged**